modules/xmpp/JingleSessionPC.js

```diff
@@ -2512,12 +2520,24 @@ export default class JingleSessionPC extends JingleSession {
             );
         const removedAnySSRCs = sdpDiffer.toJingle(remove);
 
+        // context a common object for one run of ssrc update (source-add and source-remove) so we can match them if we
+        // need to
+        const ctx = {};
```

Comment:
what is the exact problem that `ctx` solves? It's hard to follow and passing context objects is an unusual pattern. I'm wondering if there's another of fixing that?

Reply:
If you do source add and source remove in one pass to send just one presence